rsp/xxx_rest_api_design.md

```diff
@@ -0,0 +1,261 @@
+- Feature name: Lycan REST API v1
+- Start Date: 2016-03-09
+- RSP number: (leave this empty)
+
+Lycan REST API v1
+=================
+
+API base:
+
+/api/v1
+
+All methods are authenticated using a Access-Token HTTP header.
+
+A request without the proper header, or with invalid header will return an error code
+401: Unauthorized
+
+XXX: How to get the access token?
+
+## GET /maps
+
+Use this method to get available maps
+
+XXX: What is the data associated with a map?
+
+### Returns
+
+```javascript
+[
+{
+    "id": 0
+},
+{
+    "id": 1
+}
+]
+```
+
+## GET /maps/:id/instances
+
+Get running instances for a particular map
+
+### Returns
+
+```javascript
+[
+{
+    "id": 2,
+    "map": 0,
+    "uptime": 1000
+},
+{
+    "id": 3,
+    "map": 0,
+    "uptime": 30
+}
+]
+```
+
+## GET /instances/:id/entities
+
+Get entities in a particular instance
+
+Parameters:
+* `type`: Restricts search to the specified type ("monster" or "player")
```

Comment:
I haven't thought of NPCs for the API. I think it raises the broader question of how we will define and represent a NPC in-game (will those just be monster subtypes, or handled completely differently).

And I think the ideal will be to do a filtering as powerful as possible, allowing to have multiple parameters as suggested `?type=monster,npc` and even maybe equivalent of `?type=!player` if it is useful.